static/app/components/globalDrawer/components.tsx

```diff
@@ -0,0 +1,110 @@
+import {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import type {DrawerOptions} from 'sentry/components/globalDrawer/types';
+import SlideOverPanel from 'sentry/components/slideOverPanel';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+export interface DrawerPanelProps extends Required<Omit<DrawerOptions, 'onOpen'>> {
+  children: React.ReactNode;
+  isOpen?: boolean;
+  onOpen?: () => void;
+}
+
+export function DrawerPanel({
+  children,
+  isOpen = false,
+  onClose,
+  onOpen,
+  closeOnOutsideClick = true,
+  closeOnEscapeKeypress = true,
+}: DrawerPanelProps) {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(isOpen);
+  const panelRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(panelRef, () => {
+    if (isOpen && closeOnOutsideClick) {
+      onClose();
+      setIsDrawerOpen(false);
+    }
+  });
+
+  const handleEscapePress = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && isOpen && closeOnEscapeKeypress) {
+        onClose();
+        setIsDrawerOpen(false);
+      }
+    },
+    [isOpen, closeOnEscapeKeypress, onClose]
+  );
```

Comment:
We have a `useHotkeys` hook you can use here instead of manually setting up this listener.

The modal probably should use it too if it's not

Like this https://github.com/getsentry/sentry/blob/6e3cddfd060aa6db961af54197718331cc7a531b/static/app/views/settings/components/settingsSearch/index.tsx#L15

Reply:
This probably can be in the `GlobalDrawer` component also btw once you get rid of the state in here

Reply:
Updated! and removed state from the style components.